src/main/java/org/prebid/server/bidder/metax/MetaxBidder.java

```diff
@@ -0,0 +1,165 @@
+
+public class MetaxBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpMetax>> METAX_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String PUBLISHER_ID_MACRO = "{{publisherId}}";
+    private static final String AD_UNIT_MACRO = "{{adUnit}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public MetaxBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpMetax extImpMetax = parseImpExt(imp);
+                httpRequests.add(BidderUtil.defaultRequest(prepareBidRequest(request, imp),
+                        resolveEndpoint(extImpMetax),
+                        mapper));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpMetax parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), METAX_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private static BidRequest prepareBidRequest(BidRequest bidRequest, Imp imp) {
+        return bidRequest.toBuilder()
+                .imp(Collections.singletonList(modifyImp(imp)))
+                .build();
+    }
+
+    private static Imp modifyImp(Imp imp) {
+        final Banner banner = imp.getBanner();
+        final Integer width = banner != null ? banner.getW() : null;
+        final Integer height = banner != null ? banner.getH() : null;
+        if (width != null && height != null) {
+            return imp;
+        }
+
+        final List<Format> formats = banner != null ? banner.getFormat() : null;
+        if (CollectionUtils.isEmpty(formats)) {
+            return imp;
+        }
+
+        final Format firstFormat = formats.getFirst();
+        return imp.toBuilder()
+                .banner(banner.toBuilder()
+                        .w(zeroIfFormatMeasureNull(firstFormat, Format::getW))
+                        .h(zeroIfFormatMeasureNull(firstFormat, Format::getH))
+                        .build())
+                .build();
+    }
+
+    private static Integer zeroIfFormatMeasureNull(Format format, Function<Format, Integer> measureExtractor) {
+        return Optional.ofNullable(format)
+                .map(measureExtractor)
+                .orElse(0);
+    }
```

Comment:
please get rid of sending function, you can use just `Optiona.ofNullable(firstformat).map(Format::getW).orElse(0)`